lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp

```diff
@@ -168,6 +168,267 @@ class ConvertAtenEmbeddingOp : public OpConversionPattern<AtenEmbeddingOp> {
+//     for j in range(indices_length):     <- dim1
+//         for k in range(embedding_size): <- dim2
+//             if(offsets[i] <= j and j < offsets[i+1]):
+//                 output_tensor[i][k] = output_tensor[i][k] +
+//                 weight[indices[j]][k]
+//             else:
+//                 break
+//
+// Indexing maps for linalg::Generic ops
+//
+//
+// indices_indexing_map  = (d0, d1, d2) -> (d1)
+// offset_indexing_map   = (d0, d1, d2) -> (d0)
+// output_indexing_map   = (d0, d1, d2) -> (d0, d2)
+//
+
+class ConvertAtenEmbeddingBagPaddingIdxOp
+    : public OpConversionPattern<AtenEmbeddingBagPaddingIdxOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenEmbeddingBagPaddingIdxOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op->getLoc();
+    auto context = op->getContext();
+    Value weight = adaptor.weight();
+    Value indices = adaptor.indices();
+    Value offsets = adaptor.offsets();
+    Value scale_grad_by_freq = adaptor.scale_grad_by_freq();
+    Value mode = op.mode();
+    Value sparse = op.sparse();
+    Value include_last_offset = adaptor.include_last_offset();
+
+    int64_t modeInt;
+    // TODO: handle mean and max modes
+    // sum mode
+    if (!matchPattern(mode, m_TorchConstantInt(&modeInt))) {
+      return rewriter.notifyMatchFailure(
+          op, "mode is expected to be a constant integer value.");
+    }
+
+    bool isSparse;
+    // TODO: handle sparse mode, only handling dense mode rn.
+    if (!matchPattern(sparse, m_TorchConstantBool(&isSparse))) {
+      return rewriter.notifyMatchFailure(
+        op, "sparse is expected to be a constant boolean value."
+      );
+    }
+
+    // get the Embedding dimension
+    Value embeddingDim = getDimOp(rewriter, loc, weight, 1);
+
+    auto weightTy = weight.getType().cast<RankedTensorType>();
+    if (weightTy.getRank() != 2)
+      return rewriter.notifyMatchFailure(op, "weight must be rank 2");
+
+    auto indicesTy = indices.getType().cast<RankedTensorType>();
+    if (indicesTy.getRank() != 1)
+      return rewriter.notifyMatchFailure(op, "indices must be a vector");
+
+    auto offsetsTy = offsets.getType().cast<RankedTensorType>();
+    if (offsetsTy.getRank() != 1)
+      return rewriter.notifyMatchFailure(op, "offsets much be a vector");
+
+    Type weightElemTy = weightTy.getElementType();
+
+    if(!isSparse){
+      if (modeInt == 0) {
+
+        int64_t iterationMapDimension = weightTy.getRank() + indicesTy.getRank();
+
+        SmallVector<AffineExpr> indicesExpr;
+        indicesExpr.push_back(mlir::getAffineDimExpr(1, context));
+
+        auto indicesIndexingMap =
+            AffineMap::get(/*dimCount=*/iterationMapDimension, /*symbolCount=*/0,
+                          indicesExpr, context);
```

Comment:
nit: indentation seems off. Make sure you are running `git clang-format HEAD~{number_of_commits_in_your_branch}`. So if your branch has 1 commit, it should be `git clang-format HEAD~1`. This will modify some files, which you can then add and commit.

Reply:
squashed and clang formatted.